nrf_security/CMakeLists.txt

```diff
@@ -18,59 +18,51 @@ endif()
 
 include(${CMAKE_CURRENT_LIST_DIR}/cmake/nrf_security_debug.cmake)
 
-if(CONFIG_BUILD_WITH_TFM)
-  nrf_security_debug("TF-M is enabled, nRF Security will build as part of TF-M")
+#
+# Populate ARM_MBEDTLS_PATH
+#
+get_mbedtls_dir(ARM_MBEDTLS_PATH)
+
+# Neither pure Zephyr nor NS-build can set key ID encodes owner
+set(MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER       False)
+
+if(NOT CONFIG_BUILD_WITH_TFM)
+  nrf_security_debug("Building for pure Zephyr")
+
+  # Pure Zephyr requires build of Zephyr APIs
+  set(COMPILE_PSA_APIS                              True)
+
+  set(MBEDTLS_PLATFORM_EXIT_ALT                     True)
 
+else()
   # TF-M is enabled and we are not inside the TF-M build system.
-  # Thus, do not build regular `mbedcrypto` library, instead export needed
-  # settings to the TF-M build system using the `nrf_security_target`.
+  # Thus, do not build regular `mbedcrypto` library, instead export
+  # needed settings to the TF-M build system using TFM_CMAKE_OPTIONS.
 
   # NS-build: PSA APIs are already compiled in TF-M image
   set(COMPILE_PSA_APIS                              False)
 
-  # NS-build can't set key ID encodes owner
-  set(MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER       False)
-
-  # Add a library to propagate configurations and paths to TF-M
-  add_custom_target(nrf_security_target)
```

Comment:
I don't know all the details on the reasoning behind the nrf_security_target, but I am curious how (and if) this can be removed.

Also curious how this will run in the dual execution of the nrf_security (once in NSPE with zephyr build system, once inside TFM/SPE build without zephyr build system)

Reply:
I assume this can be cleaned up as @SebastianBoe proposes 👍 

A bit of history.
The reason for this was because originally we could not pass TF-M CMake options directly to TF-M build, but only through Kconfig.
Notice this code is from March 11 (PR: #426, March 9):
https://github.com/nrfconnect/sdk-nrfxlib/commit/ab66d4ac69c14158d5b143a9a1a49ef84aa75b23

and the TFM_CMAKE_OPTIONS came in place in May 7:
https://github.com/zephyrproject-rtos/zephyr/pull/34868

In the Kconfig indirection initial code, it was passed like this:
https://github.com/nrfconnect/sdk-nrfxlib/blob/ab66d4ac69c14158d5b143a9a1a49ef84aa75b23/nrf_security/Kconfig#L1665-L1667

thus the reason for `nrf_security_target`.

So I think this can and should be cleaned up.

Reply:
maybe part of the decription I just provided above would be useful in the commit message as to why this code was looking like this, and why it can now bbe cleaned up.

Can help the next one going through commit history / blame to understand the history behind this cleanup.

Reply:
Excellent to be able to remove complex generator expressions and additional targets!

Looking forward to trying to clean the code up more and make it more readable!